doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,279 @@
+
+We recommend you include the variable to be solved for as the second argument for either function. While this is optional for equations with a single symbol, it is a good practice because it ensures SymPy will solve for the desired symbol. For example, you may expect the following to solve for $x$, and SymPy will solve for $y$:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y)
+[{y: x**2}]
+```
+
+Specifying the variable to solve for ensures that SymPy solves for it:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y, x)
+[-sqrt(y), sqrt(y)]
+```
+
+## Using {func}`~.solve`
+
+- produces various output formats depending on the answer
+    - unless you use `dict=True` to ensure the result will be formatted as a dictionary, which we recommend if you want to extract information from the result programmatically
+
+You can solve an equation using {func}`~.solve` in several ways.
+
+### Make your equation into an expression that equals zero
+
+Use the fact that any expression not in an `Eq` (equation) is automatically assumed to equal zero (0) by the solving functions. You can rearrange the equation $x^2 = y$ to $x^2 - y = 0$, and {func}`~.solve` that expression. This approach is convenient if you are interactively solving an equation which can easily be rearranged to $expression = 0$.
+    
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x, y
+>>> solve(x**2 - y, x)
+[-sqrt(y), sqrt(y)]
+```
+    
+### Put your equation into `Eq` form
+
+Put your equation into `Eq` form, then apply {func}`~.solve` to the `Eq`. This approach is convenient if you are interactively solving an equation which cannot easily be rearranged to $expression = 0$.
+    
+```py
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x, y
+>>> eqn = Eq(x**2, y)
+>>> solve(eqn, x)
+[-sqrt(y), sqrt(y)]
+```
+    
+### Parse a string representing the equation
+
+Parse a string representing the equation into a form that SymPy can understand (`Eq` form), then apply {func}`~.solve` to the parsed expression.  This approach is convenient if you are programmatically reading in a string. We [recommend against using parsing a string if you are creating the expression yourself](https://github.com/sympy/sympy/wiki/Idioms-and-Antipatterns#strings-as-input). Parsing an equation from a string requires you to use [transformations](https://docs.sympy.org/dev/modules/parsing.html?highlight=parse_expr#sympy.parsing.sympy_parser.parse_expr) for SymPy to handle equals signs and create symbols from your variables.
+
+You should always include the variable to solve for if you want to extract results programmatically, to ensure that SymPy solves for the desired variable. To ensure SymPy will produce results in a consistent format, use `dict=True`. To extract the solutions, you can iterate through the list of dictionaries:  
```

Comment:
The other advantage of using `dict=True` is that it works nicely with `subs`. This is mentioned above but there are no examples. This is important because it is really the main purpose in having a `solve` function at all.

Reply:
Added an example of `subs` using critical points.